workflows/iuf/operations/add-product-to-product-catalog.yaml

```diff
@@ -0,0 +1,49 @@
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: add-product-to-product-catalog
+spec:
+  entrypoint: main
+  templates:
+### Main Steps ###
+  - name: main
+    inputs:
+      parameters:
+      - name: global_params
+    steps:
+    - - name: update-product-catalog
+        templateRef:
+          name: update-product-catalog-template
+          template: catalog-update
+        arguments:
+          parameters:
+          - name: product-name
+            value: "{{=jsonpath(inputs.parameters.global_params, '$.product_manifest.current_product.manifest.name')}}"
+          - name: product-version
+            value: "{{=jsonpath(inputs.parameters.global_params, '$.product_manifest.current_product.manifest.version')}}"
+          - name: yaml-content
+            value: "{}"
```

Comment:
I assume you tested that this works already with the `cray-product-catalog-update` container?

One case I'd be curious about is when there is already data in the product catalog for a specific product version. I'd want to make sure this doesn't overwrite it with an empty dict. I assume not, the way it does a deep merge, but I just want to make sure.

Reply:
Yeah, I was able to test the container behavior locally: https://gist.github.hpe.com/eli-kamin/78942b1d17e518e5ce06f6ea39fbd17e

However I haven't yet tested this via IUF so (in addition to the couple potentially issues the comments mention) I am not positive yet that the string is going to be formatted correctly when passed through argo template parameters.

Reply:
I was able to test this more comprehensively today and it works as expected.